mat/matrix_test.go

```diff
@@ -693,3 +707,87 @@ func TestDoer(t *testing.T) {
 		}
 	}
 }
+
+func TestMulVecTo(t *testing.T) {
+	const tol = 1e-15
+	type MulVecToer interface {
+		Matrix
+		MulVecTo(dst *VecDense, trans bool, x Vector)
+	}
+	rnd := rand.New(rand.NewSource(1))
+	random := func(n int) []float64 {
+		d := make([]float64, n)
+		for i := range d {
+			d[i] = rnd.NormFloat64()
+		}
+		return d
+	}
+	for _, a := range []MulVecToer{
+		NewBandDense(1, 1, 0, 0, random(1)),
+		NewBandDense(3, 1, 0, 0, random(1)),
+		NewBandDense(3, 1, 1, 0, random(4)),
+		NewBandDense(1, 3, 0, 0, random(1)),
+		NewBandDense(1, 3, 0, 1, random(2)),
+		NewBandDense(7, 10, 0, 0, random(7)),
+		NewBandDense(7, 10, 2, 3, random(42)),
+		NewBandDense(10, 7, 0, 0, random(7)),
+		NewBandDense(10, 7, 2, 3, random(54)),
+		NewBandDense(10, 10, 0, 0, random(10)),
+		NewBandDense(10, 10, 2, 3, random(60)),
+		NewSymBandDense(1, 0, random(1)),
+		NewSymBandDense(3, 0, random(3)),
+		NewSymBandDense(3, 1, random(6)),
+		NewSymBandDense(10, 0, random(10)),
+		NewSymBandDense(10, 1, random(20)),
+		NewSymBandDense(10, 4, random(50)),
+	} {
+		var aDense Dense
+		aDense.CloneFrom(a)
+
+		r, c := a.Dims()
+
+		for _, trans := range []bool{false, true} {
+			m, n := r, c
+			if trans {
+				m, n = c, r
+			}
+
+			for xType := 0; xType < 4; xType++ {
```

Comment:
I wonder if you could use one of the functions in list_test.go instead of this (testOneInputFunc may be the most appropriate?). testTwoInput assumes that the receiver is being modified, so I think you have to use testTwoInput, and then loop over the different receiver cases. Let me know if you have questions, as I know it's a bit of a doozy, but there are actually more like 6 different cases to test

Reply:
Yes, I planned to tackle the test after finishing `MulVecTo`.

Btw, "doozy", nice word, never seen it before.

Reply:
I've tried `testTwoInput` but for me it's simpler and faster to write the test manually.